### PR TITLE
Adds uid to `PresentTag`

### PR DESCRIPTION
**Pull request by TheBlubb14**
I could not test the code in line 593-594
I am not sure if my changes are needed there.

https://github.com/TheBlubb14/LegoDimensions/blob/main/LegoDimensions/LegoPortal.cs#L593

```cs
Array.Copy(message.Payload, i * 2 + 4, uid, 0, uid.Length);
PresentTag presentTag = new PresentTag((Pad)(message.Payload[i * 2] >> 4), (TagType)message.Payload[i * 2 + 1], (byte)(message.Payload[i * 2] & 0xF), uid);
```

### CODE REVIEW

File: LegoDimensions/LegoPortal.cs [hunk 1]
```diff
@@ -587,9 +587,11 @@ private void ReadThread(object? obj)
                             else if (message.MessageCommand == MessageCommand.None && cmdId != null && cmdId.MessageCommand == MessageCommand.TagList)
                             {
                                 _presentTags.Clear();
+                                byte[] uid = new byte[7];
                                 for (int i = 0; i < message.Payload.Length / 2; i++)
                                 {
-                                    PresentTag presentTag = new PresentTag((Pad)(message.Payload[i * 2] >> 4), (TagType)message.Payload[i * 2 + 1], (byte)(message.Payload[i * 2] & 0xF));
+                                    Array.Copy(message.Payload, i * 2 + 4, uid, 0, uid.Length);
+                                    PresentTag presentTag = new PresentTag((Pad)(message.Payload[i * 2] >> 4), (TagType)message.Payload[i * 2 + 1], (byte)(message.Payload[i * 2] & 0xF), uid);
```
Comment (Ellerbach) [inline LegoDimensions/LegoPortal.cs]: I don't think you are receiving the UID from this command. See here: https://github.com/Ellerbach/LegoDimensions/blob/main/LegoDimensionsProtocol.md#list-tags-0xd0

It means that in this case, you have to get it another way from the UID stored in _padTag by getting it from the ID.

Comment (Ellerbach) [inline LegoDimensions/LegoPortal.cs:594]: I don't think you are receiving the UID from this command. See here: https://github.com/Ellerbach/LegoDimensions/blob/main/LegoDimensionsProtocol.md#list-tags-0xd0

It means that in this case, you have to get it another way from the UID stored in _padTag by getting it from the ID.

**Comment by Ellerbach** [comment]
> I am not sure if my changes are needed there.

See my comment in the PR review ;-) You need to get it from the internal list.

**Comment by Ellerbach** [comment]
And thanks for the PR :-)

**Comment by TheBlubb14** [comment]
> > I am not sure if my changes are needed there.
> 
> See my comment in the PR review ;-) You need to get it from the internal list.

Thanks for the hint. Now I got it to work :)

**Comment by Ellerbach** [comment]
Looks good! May I abuse and ask you to adjust the test application to display the UUID? so adding the UUID in the few lines below this one: https://github.com/Ellerbach/LegoDimensions/blob/ce93166e54394cea51894b9a345dd323cfc047ec/TestLegoDimensions/Program.cs#L132

And then, it will be perfect :-D

**Review by Ellerbach — APPROVED** [review]
Thanks, it does look perfect now!

**Pull request merged**